src/Storages/StorageMerge.cpp

```diff
@@ -364,6 +365,18 @@ void StorageMerge::read(
     const size_t max_block_size,
     size_t num_streams)
 {
+    if (query_info.prewhere_info)
+    {
+        auto storage = getFirstTable([](const auto & table) { return !table->supportsPrewhere(); });
```

Comment:
This line of code duplicates `tableSupportsPrewhere`, let's unify.

Reply:
It's different. `tableSupportsPrewhere` is used to check if `optimize_move_to_prewhere` is allowed while this line of code is to check if explicit PREWHERE is valid. I'll remove `tableSupportsPrewhere` and move the implementation into `canMoveConditionsToPrewhere` instead.